File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        
        if(!root)
            return {};
        
        std::queue<TreeNode *> q;
        vector<double> result;
        
        q.push(root);
        
        while ( !q.empty() ) {
            int lSize = q.size();
            double lSum = 0;
            
            for(int i = 0; i < lSize; i++) {
                TreeNode *tmp = q.front();
                q.pop();
                
                lSum += tmp->val;
                if(tmp->left)
                    q.push(tmp->left);
                if(tmp->right)
                    q.push(tmp->right);
            }
            
            result.push_back(lSum / lSize);
        }
        
        return result;
    }
};

File: 637-average-of-levels-in-binary-tree/637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def averageOfLevels(self, root):
        """
        :type root: TreeNode
        :rtype: List[float]
        """
        if root is None:
            return []
        
        result = []
        
        q = collections.deque([root])
        
        while len(q) > 0:
            numNode = len(q)
            sum = 0.0
            for _ in range(numNode):
                node = q.popleft()
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
                sum += node.val
            result.append(sum/numNode)
            
        return result
        